src/esp/bindings_js/modules/test_page.js

```diff
@@ -75,6 +79,41 @@ class TestPage {
       this.expect(trans.y() < dropPos.y(), "trans.y() < dropPos.y()");
     }
 
+    console.log("Testing raycast");
+    // add an object
+    let banana1id = sim.addObjectByHandle(this.bananaHandle, null, "", 0);
+    sim.setTranslation(new Module.Vector3(0, 0.2, 9), banana1id, 0);
+
+    // point a ray at the object, and another ray in the opposite direction
+    let origin1 = new Module.Vector3(2, 0.2, 9);
+    let direction1 = new Module.Vector3(-1, 0, 0);
+    let ray1 = new Module.Ray(origin1, direction1);
+    let origin2 = new Module.Vector3(-4, 0.2, 9);
+    let direction2 = new Module.Vector3(1, 0, 0);
+    let ray2 = new Module.Ray(origin2, direction2);
+
+    // check that castRay returns the object and that the hit location is correct
+    let hit = sim.castRay(ray1, 6, 0).hits.get(0);
+    this.expect(hit.objectId == banana1id);
+    this.expect(Math.abs(hit.point.x() - 0.0172938) < 0.001);
+    this.expect(Math.abs(hit.point.y() - 0.2) < 0.001);
+    this.expect(Math.abs(hit.point.z() - 9.0) < 0.001);
```

Comment:
Maybe these could be compacted into a single function roughly like: `(vector1-vector2).magnitude() < epsilon`?

Reply:
Yep this makes sense and has been done, thanks!